pkg/cmd/clean/exec.go

```diff
@@ -0,0 +1,140 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package init
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	clustermanagerclient "open-cluster-management.io/api/client/operator/clientset/versioned"
+	"open-cluster-management.io/clusteradm/pkg/helpers"
+	"open-cluster-management.io/clusteradm/pkg/helpers/apply"
+
+	"github.com/spf13/cobra"
+
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+)
+
+func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
+	klog.V(1).InfoS("clean options:", "dry-run", o.ClusteradmFlags.DryRun, "output-file", o.outputFile)
+	o.values = Values{
+		Hub: Hub{
+			TokenID:     helpers.RandStringRunes_az09(6),
+			TokenSecret: helpers.RandStringRunes_az09(16),
+		},
+	}
+	return nil
+}
+
+func (o *Options) validate() error {
+	restConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	apiExtensionsClient, err := apiextensionsclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+	installed, err := helpers.IsClusterManagerInstalled(apiExtensionsClient)
+	if err != nil {
+		return err
+	}
+	if !installed {
+		return fmt.Errorf("hub not be initialized")
+	}
+	return nil
+}
+
+func (o *Options) run() error {
+	output := make([]string, 0)
+
+	//Clean ClusterManager CR resource firstly
+	f := o.ClusteradmFlags.KubectlFactory
+	config, err := f.ToRESTConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+	clusterManagerClient, err := clustermanagerclient.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = clusterManagerClient.OperatorV1().ClusterManagers().Delete(context.Background(), "cluster-manager", metav1.DeleteOptions{})
+	if err != nil {
+		log.Fatal(err)
+	}
+	if !o.ClusteradmFlags.DryRun {
+		b := retry.DefaultBackoff
+		b.Duration = 1 * time.Minute
+
+		err = WaitResourceToBeDelete(context.Background(), clusterManagerClient, "cluster-manager", b)
+		if err != nil {
+			log.Println("Recource cluster-manager have been deleted firstly.")
+
+		}
+	}
+
+	//Clean other resources
+	kubeClient, apiExtensionsClient, _, err := helpers.GetClients(f)
+	if err != nil {
+		return err
+	}
+	kubeClient.AppsV1().Deployments("open-cluster-management").Delete(context.Background(), "cluster-manager", metav1.DeleteOptions{})
```

Comment:
I think you should delete the ClusterManager resource, and when it is removed, Delete the operator related resource

Reply:
you would not need to remove resources in open-cluster-management-hub if cluster-manager is deleted. I think you need to wait/watch until cluster-manager is cleaned at first, then delete the resources in open-cluster-management ns and also the klusterlet crd.